### PR TITLE
Integrate mpeg2 video support into fluster

### PR DESCRIPTION
**Pull request by rsanchez87**

### CODE REVIEW

**Comment by ylatuya** [comment]
This PR is adding new test suites to fluster, hence new features but none of your commits has a `feature:` type.

You should split [build: compilation: add gen_mpeg2_video.py in the absence of raw file creation](https://github.com/fluendo/fluster/pull/250/commits/8b33f3d55bcc3810b653897dfcbebeb2ee16d10d) into 2 commits, one that adds the helper script to generate the test suite, and another one that adds the test suites like:

 `feature:  add ISO IEC 13818-4 MPEG2  test suites`

**Comment by rsanchez87** [comment]
> This PR is adding new test suites to fluster, hence new features but none of your commits has a `feature:` type.
> 
> You should split [build: compilation: add gen_mpeg2_video.py in the absence of raw file creation](https://github.com/fluendo/fluster/pull/250/commits/8b33f3d55bcc3810b653897dfcbebeb2ee16d10d) into 2 commits, one that adds the helper script to generate the test suite, and another one that adds the test suites like:
> 
> `feature: add ISO IEC 13818-4 MPEG2 test suites`

Perfect, I'll split it and leave it as you commented, thanks

**Comment by ylatuya** [comment]
The README must be updated as well

**Comment by rsanchez87** [comment]
> The README must be updated as well



> The README must be updated as well



> The README must be updated as well

Yes, I know, the PR is still in Draft

File: test_suites/mpeg2v/MPEG2-VIDEO-422.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+    "name": "MPEG2-VIDEO-422",
```
[the remark this answers is not in the document]
Comment (rsanchez87) [reply test_suites/mpeg2v/MPEG2-VIDEO-422.json:2]: Same in `MPEG2_MAIN`, changed too.

File: test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+        {
+            "name": "hhi_burst_short.bits",
+            "source": "https://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_IEC_13818-4_2004_Conformance_Testing/Video/bitstreams/main-profile/hhi/hhi_burst_short/hhi_burst_short.bits.gz",
+            "source_checksum": "a68d3436d89ffc896393b18e8556a33c",
+            "input_file": "hhi_burst_short.bits",
+            "output_format": "yuv420p",
+            "result": ""
+        },
+        {
+            "name": "gi_stream",
```
[the remark this answers is not in the document]
Comment (rsanchez87) [reply test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json:231]: I don't understand this comment. Do you mean to change the name to ‘cheerleader’ in this case...? Has it been done in any test suite before?

File: test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json
[the remark this answers is not in the document]
Comment (rsanchez87) [reply test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json:1]: You are right, there are different `bitstream` files in **gi** folder

[the remark this answers is not in the document]
Comment (rsanchez87) [reply test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json:1]: see **l112** of **scripts/gen_mpeg2_video.py**

File: test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+        {
+            "name": "video.bits",
+            "source": "https://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_IEC_13818-4_2004_Conformance_Testing/Video/bitstreams/main-profile/gi/gi4/video.bits.gz",
+            "source_checksum": "bc2d7a72cbcc85d1bf6962b69e4fe7d4",
+            "input_file": "video.bits",
+            "output_format": "yuv420p",
+            "result": ""
+        },
+        {
+            "name": "conf2.bits",
```
[the remark this answers is not in the document]
Comment (rsanchez87) [reply test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json]: same of above

File: test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+        {
+            "name": "nokia7_dual.bit",
+            "source": "https://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_IEC_13818-4_2004_Conformance_Testing/Video/bitstreams/main-profile/nokia/nokia_7/nokia7_dual.bit.gz",
+            "source_checksum": "f3d04767b878df837bd9c5efaeca26ca",
+            "input_file": "nokia7_dual.bit",
+            "output_format": "yuv420p",
+            "result": ""
+        },
+        {
+            "name": "conf4.bit",
```
[the remark this answers is not in the document]
Comment (rsanchez87) [reply test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json]: same of above

File: test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+            "source": "https://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_IEC_13818-4_2004_Conformance_Testing/Video/bitstreams/main-profile/tek/Tek-5-long/conf4.bit.gz",
+            "source_checksum": "c22a3c2f94e5d246f61e9e44d4f334a5",
+            "input_file": "conf4.bit",
+            "output_format": "yuv420p",
+            "result": ""
+        },
+        {
+            "name": "test.mpg",
```
[the remark this answers is not in the document]
Comment (rsanchez87) [reply test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json]: same

File: test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdimopoulos) [reply test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json]: In this case the test vector is https://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_IEC_13818-4_2004_Conformance_Testing/Video/bitstreams/main-profile/gi/gi_from_tape/.
Right now vector name is `gi_stream` because of the name of the file that is inside `gi_stream.gz`. Same happens with all test vectors. For gi group vectors this causes a problem because `gi6`, `gi7`, `gi_9` all have the same name bit_stream.gz.
So we need to find a way around this issue.

This hasn't been done on any other test suite before because we didn't have this problem.

[the remark this answers is not in the document]
Comment (rsanchez87) [reply test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json]: I think with last commit (https://github.com/fluendo/fluster/pull/250#discussion_r1955800968) is already solved

File: test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (mdimopoulos) [reply test_suites/mpeg2v/MPEG2-VIDEO-MAIN.json]: This will be left as is because it doesn't clash with any other file. As discussed, there is no perfect generic solution and we don't want to add customisations for each vector.

**Review by mdimopoulos — APPROVED** [review]
